File: src/main/java/com/rg/webmvctest/domain/RegistrationService.java
package com.rg.webmvctest.domain;

import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class RegistrationService {

    private final UserRegistrationRepository userRegistrationRepository;

    public RegistrationService(UserRegistrationRepository userRegistrationRepository) {
        this.userRegistrationRepository = userRegistrationRepository;
    }

    public UserRegistration registerUser(String emailAddress, String name, String lastName) {
        return userRegistrationRepository.insert(emailAddress, name, lastName);
    }

    public UserRegistration updateRegistration(UserRegistration userRegistration) {
        return userRegistrationRepository.update(userRegistration);
    }

    public UserRegistration findUserRegistrationById(String registrationId) {
        return userRegistrationRepository.findByEmailAddress(registrationId);
    }

    public List<UserRegistration> findAllUserRegistrations() {
        return userRegistrationRepository.findAll();
    }
}

File: src/main/kotlin/com/rg/webmvctest/domain/RegistrationService.kt
package com.rg.webmvctest.domain

import org.springframework.stereotype.Component

@Component
open class RegistrationService(
        private val userRegistrationRepository: UserRegistrationRepository
) {

    open fun registerUser(emailAddress: String, name: String, lastName: String): UserRegistration =
            userRegistrationRepository.insert(emailAddress, name, lastName)

    open fun updateRegistration(userRegistration: UserRegistration): UserRegistration =
            userRegistrationRepository.update(userRegistration)

    open fun findUserRegistrationById(registrationId: String): UserRegistration? =
            userRegistrationRepository.findByEmailAddress(registrationId)

    open fun findAllUserRegistrations(): List<UserRegistration> =
            userRegistrationRepository.findAll()
}